src/Backend/OnboardingInsuranceAPI/Areas/Insurance/GetContribution.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.Enums;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class GetContribution : IHandler
+{
+    private readonly DataContext _context;
+
+    public GetContribution(DataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<ContributionDataResponse> Handle(ContributionDataRequest requestedData, string pid)
+    {
+        if(string.IsNullOrEmpty(pid))
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+
+        var birthDate = _context.Users.FirstOrDefault(u => u.Pid == pid)?.Birthdate;
+        if(string.IsNullOrEmpty(birthDate))
+            throw new ApiException(ErrorCode.NotFound);
+
+        var carContribution = JsonSerializer.Deserialize<CarTypesContribution>(requestedData.CarType);
```

Comment:
Really think about this! You have already deserialized this class once. Why do it again?